### PR TITLE
third lab

### PR DESCRIPTION
**Pull request by maksimkha**
Третья лаба "Морской бой"

### CODE REVIEW

File: Lab3/app/build.gradle [hunk 1]
```diff
@@ -14,7 +18,7 @@ android {
         minSdkVersion 24
         targetSdkVersion 30
         versionCode 1
```
Comment (gmltA) [inline Lab3/app/build.gradle]: А тут што?

File: Lab3/app/src/main/java/com/example/battleship/AccountActivity.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            ) {
+                if (position == iCurrentSelection)
+                    return
+                Log.i("item", position.toString())
+                finalRef = if (position != 0){
+                    storageRef.child("avatar$position.jpg")
+                } else{
+                    storageRef.child("custom.jpg")
+                }
+                GlideApp.with(applicationContext)
```
Comment (gmltA) [inline Lab3/app/src/main/java/com/example/battleship/AccountActivity.kt]: Не бачу тут падтрымкі Gravatar.

File: Lab3/app/src/main/java/com/example/battleship/GameStartActivity.kt [hunk 3]
```diff
@@ -0,0 +1,503 @@
...
+    var role: String? = ""
+    var antiRole: String? = ""
+    var storageRef: StorageReference? = null
+    lateinit var table: TableLayout
+    lateinit var enemy: TableLayout
+    lateinit var startBtn: Button
+    var ships: Int = 0
+    var cellsInt: Int = 0
+    var usedCells = Array(20) { i -> Array(2) { j -> 0 } }
+    var finalCells = Array(20) { i -> Array(2) { j -> 0 } }
```
Comment (gmltA) [inline Lab3/app/src/main/java/com/example/battleship/GameStartActivity.kt]: Game logic and DB operations should be in a ViewModel.

**Review by gmltA — CHANGES_REQUESTED** [review]

File: Lab3/app/build.gradle (hunk 1 above)
Comment (maksimkha) [reply Lab3/app/build.gradle]: fixed

File: Lab3/app/src/main/java/com/example/battleship/AccountActivity.kt (hunk 2 above)
Comment (maksimkha) [reply Lab3/app/src/main/java/com/example/battleship/AccountActivity.kt]: падтрымал

File: Lab3/app/src/main/java/com/example/battleship/GameStartActivity.kt (hunk 3 above)
Comment (maksimkha) [reply Lab3/app/src/main/java/com/example/battleship/GameStartActivity.kt]: move logic and DB's to ViewModel

File: Lab3/app/build.gradle (hunk 1 above)
Comment (maksimkha) [reply Lab3/app/build.gradle]: скрины для удобства
image
image

File: Lab3/app/src/main/java/com/example/battleship/AccountActivity.kt (hunk 2 above)
Comment (maksimkha) [reply Lab3/app/src/main/java/com/example/battleship/AccountActivity.kt]: image

File: Lab3/app/src/main/java/com/example/battleship/GameStartActivity.kt (hunk 3 above)
Comment (maksimkha) [reply Lab3/app/src/main/java/com/example/battleship/GameStartActivity.kt]: image
image

**Review by gmltA — APPROVED** [review]

**Pull request merged**